grafana/resource_amixr_integration.go

```diff
@@ -0,0 +1,393 @@
+* [HTTP API](https://grafana.com/docs/grafana-cloud/oncall/oncall-api-reference/)
+`,
+
+		Create: resourceIntegrationCreate,
+		Read:   resourceIntegrationRead,
+		Update: resourceIntegrationUpdate,
+		Delete: resourceIntegrationDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+				Description:  "The name of the service integration",
+			},
+			"team_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The id of the team",
+			},
+			"type": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice(integrationTypes, false),
+				ForceNew:     true,
+				Description:  "The type of integration",
+			},
+			"default_route": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"escalation_chain_id": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"slack": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"channel_id": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+								},
+							},
+							MaxItems: 1,
+						},
+					},
+				},
+				MaxItems:    1,
+				Description: "The Default route for all alerts from the given integration",
+			},
+			"link": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The link for using in an integrated tool.",
+			},
+			"templates": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"resolve_signal": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"grouping_key": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"slack": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"title": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"message": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"image_url": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+								},
+							},
+							MaxItems: 1,
+						},
+					},
+				},
+				MaxItems:    1,
+				Description: "Jinja2 templates for Alert payload.",
+			},
+		},
+	}
+}
+
+func resourceIntegrationCreate(d *schema.ResourceData, m interface{}) error {
```

Comment:
I think these should be prefixed as well, to be more explicit when autocompleting

```suggestion
func resourceAmixrIntegrationCreate(d *schema.ResourceData, m interface{}) error {
```